### ISSUE TITLE
Water purifying bugs

### ISSUE THREAD

**Issue by ontrigger** [loiste-interactive/Obenseuer-Issues#718]
Having a bottle of clean water in a chemical lab inventory can lead to some bugs.

Bug 1:
If you try to purify water and the attempt fails, then the clean water bottle will disappear and get stuck in purgatory until you get a successful purification attempt, in which case it will reappear

Bug 2:
If you have two almost-full clean bottles like this and successfully purify
![image](https://github.com/loiste-interactive/Obenseuer-Issues/assets/8556074/334d99a0-af3d-4690-b13e-f97137d31af9)
Upon clicking purify, one of the bottles that is not removed from the inventory will overfill like this
![image](https://github.com/loiste-interactive/Obenseuer-Issues/assets/8556074/9ac402cd-91a3-46da-9ea1-85e4942a4fc4)
If you then remove the overfilled bottle, a bottle with negative volume will appear after crafting is finished
![image](https://github.com/loiste-interactive/Obenseuer-Issues/assets/8556074/abd8c1ed-68ce-4ed4-be27-35efb31077fb)

**Comment by CVariable** [comment]
This is still happening in 3.22, crafting mushroom extract multiple times always results in the previous liquid being deleted in favor of the new one. Crafting extract while already having some will take the extract container, empty it and replace it with the new liquid, glowing bottoms in this case.

**Comment by greensalt99** [comment]
0.3.33 i guess same bug. had:
 partially filled water bottle+ empty canister +dirty water canister. used purify water.  = result, water is gone, lot of sewage.
 tried again. sewage is gone . have more clean water. 
 tried many times and always some mess with water or sewage containers. used empty bottle stack mix and canisters.
 this might confuse script. 
 
![chem lab chaos](https://github.com/loiste-interactive/Obenseuer-Issues/assets/154023649/5664d81f-994e-4692-82ba-557359c7d05d)

 i suggest fix. script checks if bottles/canisters have space for success and fail filling. 
 if success will clean water have space in water containers/bottles/canisters. condition empty of filled with pure water.
 if fail will sewage water have space in containers/bottles/canisters. condition empty or filled with sewage.
 
 so basically this check will decide where crafting result go and take planed action correctly.

**Comment by Oskutin** [comment]
Closing this as this is very old issue and the underlying system has been changed
